### PR TITLE
Add an apache::vhost::proxy define

### PR DESCRIPTION
**Pull request by wbclark**
This builds on apache::vhost::fragment with a more specialized type for
a reverse proxy setup. This reuses the same template as apache::vhost
but has data types. The types are stricter, but this leads to a clearer
API. For example, it doesn't accept String and Array[String] but in that
case always wants an Array.

The old parameters on vhost remain and can continue to be used.

### CODE REVIEW

**Comment by wbclark** [comment]
Discussed with @ekohl and agreed to take over https://github.com/puppetlabs/puppetlabs-apache/pull/2167

This PR therefore contains my latest work on it to get the tests into a working state.

File: manifests/vhost/proxy.pp
Comment (ekohl) [inline manifests/vhost/proxy.pp:30]: This should be updated. I initially copied it from `apache::vhost`. I'd now also make it a separate `@example` block.

File: types/vhost/proxypass.pp
Comment (ekohl) [inline types/vhost/proxypass.pp:1]: In the define itself there's more documentation. That probably makes sense in this file as well (or only here?)

**Comment by david22swan** [comment]
@wbclark Apologies for the wait on a review.
Anyway, after taking a look over this pr it for the most part looks good to me, however would be more comfortable if you responded to ekohls comments before I merged it in.

**Comment by wbclark** [comment]
Thanks very much @ekohl and @david22swan for the feedback, and sorry for the delay in resuming work on this.

I've updated this with major improvements to the documentation for the data type and the defined resource type, in order to address the provided feedback.

**Comment by wbclark** [comment]
@david22swan thanks, I've made the requested updates.

I did these in separate commits for easier reviewing, I assume you'll want to squash before merging. Let me know if you prefer me to do that and force push, or if you'd rather do the squashing on your end during merge. Thanks!

**Comment by david22swan** [comment]
@wbclark One more quick change sorry, but to the best of my knowledge, $priority should be `Variant[Integer,String,Boolean]`
Integer and String for passing in input and Boolean for if you wish to disable it and set it to false

**Comment by david22swan** [comment]
@wbclark Thanks for putting in the work and getting this so clean. It LGTM now so I'm gonna go ahead and merge.

Also, to clarify about the accepted values, while the apache config only accepts integer values being given, the original examples and default variable both indicate it being passed through as a string, thus necessitating that we accept input in that manner.

Anyway, thanks again for putting in the work and getting it done so fast :)
Looking forward to seeing more

**Review by david22swan — APPROVED** [review]
LGTM

**Comment by wbclark** [comment]
Thanks @david22swan and @ekohl ! It's been a pleasure working with you on it.

**Pull request merged**
